Sources/SwiftParser/TokenSpec.swift

```diff
@@ -12,6 +12,32 @@
 
 @_spi(RawSyntax) import SwiftSyntax
 
+/// Pre-computes the keyword a lexeme might represent. This makes matching
+/// a lexeme that has been converted into `PrepareForKeyword` match cheaper to
+/// match against multiple `TokenSpec` that assume a keyword.
```

Comment:
This would remove the main advantage I see in not just converting to keyword in the lexer - ie. for identifier cases, we don't need to do the string -> keyword conversion. But with `PrepareForKeywordMatch` we're now ignoring that. So why not just always convert to a keyword in the lexer once and simplify the rest?

Reply:
My impression is that the entire “how should we represent things that might be keywords in the lexer” discussion is related to this PR but it’s fundamentally an orthogonal issue. 

IIUC we all agree that `RawTokenKind` should be a trivial enum, so I would propose that we merge this PR and continue the discussion of how keywords should be represented in a follow-up PR. WDYT?

Reply:
To be clear, I'm not arguing against merging this PR. I was mostly responding to the comment in the other PR about this addressing it 😅